src/ranch.erl

```diff
@@ -52,10 +52,17 @@
 -type opts() :: any() | transport_opts(any()).
 -export_type([opts/0]).
 
+-type alarm() :: undefined
+	| #{treshold => non_neg_integer(),
+	    callback => fun((ref(), pid(), [pid()]) -> any()) | {module(), atom()},
+	    cooldown => timeout()}.
+-export_type([alarm/0]).
+
 -type transport_opts(SocketOpts) :: #{
 	connection_type => worker | supervisor,
 	handshake_timeout => timeout(),
 	max_connections => max_conns(),
+	alarm => alarm(),
```

Comment:
Would there be other interesting alarms in the future? In which case the name needs to be changed.

Reply:
I have been thinking about it for a while, and I don't think there are any. This is, give or take, the only really dynamic part in ranch, everything else is done via setup/configuration by the user. Number of file handles (`emfile` etc) _may_ in theory be something we could keep an eye on and warn about, but I'm not sure it would be useful.

Reply:
Like we discussed it's fine to deprecate/rename later anyway.